server/pulp/server/managers/content/query.py

```diff
@@ -290,7 +290,14 @@ def request_content_unit_file_path(self, content_type, relative_path):
         unit_path = os.path.join(self.get_root_content_dir(content_type), relative_path)
         unit_dir = os.path.dirname(unit_path)
         if not os.path.exists(unit_dir):
-            os.makedirs(unit_dir)
+            try:
+                os.makedirs(unit_dir)
+            except OSError, e:
+                # In case two workers are syncing exact same content simultaneously and the
+                # directory got created in between the above check and now. See issue #902.
+                if e.errno != 17:
```

Comment:
The `17` should be replaced with a name provided by the python standard lib `errno`. Different systems may number these errors differently, but Python guarantees that the names provided by `errno` will map correctly on all systems Python runs on. See how I did this [in this other PR](https://github.com/pulp/pulp/pull/1791/files#diff-6ceaf55d8c7a51c3a8243b1d577dcd11R96).